utils/errors.ts

```diff
@@ -1,33 +1,38 @@
-import { formatJSON } from ".";
-
-export const getIssueUpdateError = (
-    resource: "state" | "description" | "title" | "assignee",
-    data: { number: number; id: string; team: { key: string } },
-    syncedIssue: { githubIssueNumber: number; githubIssueId: number },
-    updatedIssueResponse: { statusCode: number; json: () => any }
-): string => {
-    return `Failed to update GitHub issue ${resource} for ${data.team.key}-${
-        data.number
-    } [${data.id}] on GitHub issue #${syncedIssue.githubIssueNumber} [${
-        syncedIssue.githubIssueId
-    }], received status code ${
-        updatedIssueResponse.statusCode
-    }, body of ${formatJSON(updatedIssueResponse.json())}.`;
-};
-
-export const getOtherUpdateError = (
-    resource: "comment" | "attachment",
-    data: { number: number; id: string; team: { key: string } },
-    createdIssue: { number: number; id: number },
-    createdIssueResponse: { statusCode: number; json: () => any },
-    responseBody: any
-): string => {
-    return `Failed to update GitHub issue ${resource} for ${data.team.key}-${
-        data.number
-    } [${data.id}] on GitHub issue #${createdIssue.number} [${
-        createdIssue.id
-    }], received status code ${
-        createdIssueResponse.statusCode
-    }, body of ${formatJSON(responseBody)}.`;
-};
-
+import { formatJSON } from ".";
+
+export const getIssueUpdateError = (
+    resource: "state" | "description" | "title" | "assignee",
+    data: { number: number; id: string; team: { key: string } },
+    syncedIssue: { githubIssueNumber: number; githubIssueId: number },
+    updatedIssueResponse: { statusCode: number; json: () => any }
+): string => {
+    return `Failed to update GitHub issue ${resource} for ${data.team.key}-${
+        data.number
+    } [${data.id}] on GitHub issue #${syncedIssue.githubIssueNumber} [${
+        syncedIssue.githubIssueId
+    }], received status code ${
+        updatedIssueResponse.statusCode
+    }, body of ${formatJSON(updatedIssueResponse.json())}.`;
+};
+
+export const getOtherUpdateError = (
+    resource: "comment" | "attachment",
+    data: { number: number; id: string; team: { key: string } },
+    createdIssue: { number: number; id: number },
+    createdIssueResponse: { statusCode: number; json: () => any },
+    responseBody: any
+): string => {
+    return `Failed to update GitHub issue ${resource} for ${data.team.key}-${
+        data.number
+    } [${data.id}] on GitHub issue #${createdIssue.number} [${
+        createdIssue.id
+    }], received status code ${
+        createdIssueResponse.statusCode
+    }, body of ${formatJSON(responseBody)}.`;
+};
+
+export class ApiError extends Error {
+    constructor(public message: string, public statusCode: number) {
+        super(message);
+    }
+}

```

Comment:
New exception class used by the handlers, so they can throw instead of calling `res.status` directly.